### DISCUSSION TITLE
Passing value from 'upgrade' to 'open' handler

### DISCUSSION THREAD

**Discussion by PlainBane** [uNetworking/uWebSockets.js#322] [Q&A] [answered]
Hello,
Is there a way to do something like this:

` upgrade: (res, req, context) => {`
 `   userId = req.getFromCookiesToken();`
`    if(userId) {`
 `// Upgrade to Websocket`
`}`

` open: (ws) => {`
  `ws.uid = userId; // Passed from upgrade handler. <--- Can this be done?`
`}`

**Comment by PlainBane** [comment, marked as answer]
Found the answer. I'm blind.
`res.upgrade({url: url,` **customValueHere** `}, secWebSocketKey, secWebSocketProtocol, secWebSocketExtensions, context);`

**Comment by ghost** [comment]
Haha, great 👍
